### PR TITLE
📖 docs: updated flavors and machine pool docs

### PR DESCRIPTION
**Pull request by richardcase**
**What this PR does / why we need it**:
An update to the machine pool, eks and release docs. Also copying the templates to the output folder.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by richardcase** [comment]
@michaelbeaumont - would you be able to take a look at this?

File: docs/book/src/topics/machinepools.md [hunk 1]
```diff
@@ -28,11 +25,38 @@ clusterctl config cluster my-cluster --kubernetes-version v1.16.8 --flavor machi
...
+
+Cluster API Provider AWS (CAPA) has experimental support for [EKS Managed Node Groups](https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html) using `MachinePool` through the infrastructure type `AWSManagedMachinePool`. An `AWSManagedMachinePool` corresponds to an [AWS AutoScaling Groups](https://docs.aws.amazon.com/autoscaling/ec2/userguide/AutoScalingGroup.html) that is used for an EKS managed node group. .
+
+The AWSManagedMachinePool controller creates and manages an EKS managed node group with in turn manages an AWS AutoScaling Group of managed EC2 instance types.
+
+### Using `clusterctl` to deploy
+
+To deploy an EKS managed node group using AWSManagedMachinePool via `clusterctl config` you can use a [flavor](https://cluster-api.sigs.k8s.io/clusterctl/commands/config-cluster.html#flavors).
+
+Make sure to set up your AWS environment as described [here](https://cluster-api.sigs.k8s.io/user/quick-start.html).
```
Comment (randomvariable) [inline docs/book/src/topics/machinepools.md:40]:
```suggestion
Make sure to set up your AWS environment as described [here](../getting-started.md).
```

The getting started in this repo just includes the upstream CAPI doc.

File: docs/book/src/topics/machinepools.md [hunk 2]
```diff
@@ -28,11 +25,38 @@ clusterctl config cluster my-cluster --kubernetes-version v1.16.8 --flavor machi
...
 
 The template used for this [flavor](https://cluster-api.sigs.k8s.io/clusterctl/commands/config-cluster.html#flavors)
-is located [here](../../templates/cluster-template-machinepool.yaml). For example you can add it to `~/.cluster-api/overrides/infrastructure-aws/0.0.0/cluster-template-machinepool.yaml`
+is located [here](../../templates/cluster-template-machinepool.yaml).
+
+## AWSManagedMachinePool
+
+Cluster API Provider AWS (CAPA) has experimental support for [EKS Managed Node Groups](https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html) using `MachinePool` through the infrastructure type `AWSManagedMachinePool`. An `AWSManagedMachinePool` corresponds to an [AWS AutoScaling Groups](https://docs.aws.amazon.com/autoscaling/ec2/userguide/AutoScalingGroup.html) that is used for an EKS managed node group. .
+
+The AWSManagedMachinePool controller creates and manages an EKS managed node group with in turn manages an AWS AutoScaling Group of managed EC2 instance types.
```
Comment (michaelbeaumont) [inline docs/book/src/topics/machinepools.md:34]:
```suggestion
Cluster API Provider AWS (CAPA) has experimental support for [EKS Managed Node Groups](https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html) using `MachinePool` through the infrastructure type `AWSManagedMachinePool`. An `AWSManagedMachinePool` corresponds to such a managed node group which is in turn powered by an [AWS AutoScaling Group](https://docs.aws.amazon.com/autoscaling/ec2/userguide/AutoScalingGroup.html), that handles the orchestration of EC2 instances.
```

File: docs/book/src/topics/machinepools.md [hunk 3]
```diff
@@ -28,11 +25,38 @@ clusterctl config cluster my-cluster --kubernetes-version v1.16.8 --flavor machi
...
+
+### Using `clusterctl` to deploy
+
+To deploy an EKS managed node group using AWSManagedMachinePool via `clusterctl config` you can use a [flavor](https://cluster-api.sigs.k8s.io/clusterctl/commands/config-cluster.html#flavors).
```
Comment (michaelbeaumont) [inline docs/book/src/topics/machinepools.md:38]:
```suggestion
To deploy an EKS managed node group using `AWSManagedMachinePool` via `clusterctl config` you can use a [flavor](https://cluster-api.sigs.k8s.io/clusterctl/commands/config-cluster.html#flavors).
```

File: docs/book/src/topics/machinepools.md (hunk 2 above)
Comment (michaelbeaumont) [reply docs/book/src/topics/machinepools.md:34]: Seemed a little redundant as is

Comment (richardcase) [reply docs/book/src/topics/machinepools.md]: Changed.

File: docs/book/src/topics/machinepools.md (hunk 3 above)
Comment (richardcase) [reply docs/book/src/topics/machinepools.md]: Changed.

**Review by michaelbeaumont — APPROVED** [review]

**Comment by randomvariable** [comment]
Oops, could you rebase.

I thought I had merged this for some reason.

**Comment by richardcase** [comment]
> I thought I had merged this for some reason.

I thought it had merged as well 😄 Rebasing now....

**Pull request merged**